File: src/nav_pkg/src/wp_node.cpp
#include<ros/ros.h>
#include<std_msgs/String.h>
void NavResultCallback(const std_msgs::String::ConstPtr& msg)//指针引用
{
    ROS_INFO("[Received]: %s", msg->data.c_str());
    
}
int main(int argc, char **argv)
{
    ros::init(argc, argv, "wp_node");
    ros::NodeHandle nh;
    ros::Publisher wp_pub = nh.advertise<std_msgs::String>("/waterplus/navi_waypoint", 10);
    ros::Subscriber wp_sub = nh.subscribe("/waterplus/navi_result", 10, &NavResultCallback);
        
    ros::Rate loop_rate(1); // 1 Hz

    while (ros::ok())
    {
        std_msgs::String msg;
        msg.data = "1";
        wp_pub.publish(msg);
        ROS_INFO("Published: %s", msg.data.c_str());
        ros::spinOnce();
        loop_rate.sleep();
    }

    return 0;
}

File: src/nav_pkg/scripts/wp_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String

def PublisherWaypoint():
    navi_waypoint=String()
    navi_waypoint.data="2"

    rospy.sleep(1)
    wp_pub.publish(navi_waypoint)
    rospy.loginfo("已发布导航目标点：%s", navi_waypoint.data)
    
def NavResultCallback(msg):
    rospy.loginfo("导航结果：%s", msg.data)
if __name__ == '__main__':
    rospy.init_node("wp_node")

    wp_pub=rospy.Publisher("/waterplus/navi_waypoint",String,queue_size=10)
    wp_sub=rospy.Subscriber("/waterplus/navi_result", String, NavResultCallback,queue_size=10)

    PublisherWaypoint()#发布导航目标点
    rospy.spin()#保持节点运行，等待回调函数被调用
